File: python/leetcode/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0 or n == 1 or n == 2:
            return n

        steps = [0] * (n+1)
        steps[0] = 0
        steps[1] = 1
        steps[2] = 2

        for i in range(3, n+1):
            steps[i] = steps[i-1] + steps[i-2]

        print(steps)

        return steps[n]

File: c#/leetcode/70-climbing-stairs.cs
public class Solution
{
  public int ClimbStairs(int n)
  {
    if (n == 1 || n == 2)
    {
      return n;
    }

    int[] stairs = new int[n + 1];

    stairs[0] = 0;
    stairs[1] = 1;
    stairs[2] = 2;

    for (int i = 3; i < stairs.Length; i++)
    {
      stairs[i] = stairs[i - 1] + stairs[i - 2];
    }

    return stairs[n];
  }
}